src/main/java/co/cask/gcp/speech/SpeechToTextTransform.java

```diff
@@ -0,0 +1,332 @@
+    boolean hasTranscriptField = false;
+    // one of the output field in which the transcript will be specified must be specified
+    if (transcriptPartsField != null && !transcriptPartsField.isEmpty()) {
+      topLevel.add(transcriptWithConfidenceSchema(transcriptPartsField));
+      hasTranscriptField = true;
+    }
+
+    if (transcriptTextField != null && !transcriptTextField.isEmpty()) {
+      topLevel.add(topConfidenceSchema(transcriptTextField));
+      hasTranscriptField = true;
+    }
+
+    if (!hasTranscriptField) {
+      throw new IllegalArgumentException("Either 'Transcript Parts Field' or 'Transcript Text Field' or both must be " +
+                                           "specified.");
+    }
+
+    if (outSchema != null && outSchema.getFields() != null) {
+      topLevel.addAll(outSchema.getFields());
+    }
+    return Schema.recordOf("record", topLevel);
+  }
+
+  /**
+   * Request object from the Plugin REST call.
+   */
+  public static final class Request extends SpeechTransformConfig {
+    private Map<String, Schema> inputSchemas;
+  }
+
+  @Path("getSchema")
+  public Schema getSchema(Request request) {
+    Set<Map.Entry<String, Schema>> entries = request.inputSchemas.entrySet();
+    return getSchema(entries.iterator().next().getValue(), request.getTranscriptionPartsField(),
+                     request.getTranscriptionTextField());
+  }
+
+  private SpeechSettings getSettings() throws IOException {
+    SpeechSettings.Builder builder = SpeechSettings.newBuilder();
+    if (config.getServiceAccountFilePath() != null) {
+      builder.setCredentialsProvider(() -> GCPUtils.loadServiceAccountCredentials(config.getServiceAccountFilePath()));
+    }
+    return builder.build();
+  }
+
+  private void setRecognitionConfig() {
+    recognitionConfig = RecognitionConfig.newBuilder()
+      .setEncoding(getAudioEncoding())
+      .setSampleRateHertz(Integer.parseInt(config.sampleRate))
+      .setProfanityFilter(config.profanity.equals("true"))
+      .setLanguageCode(config.language == null ? "en-US" : config.language)
+      .build();
+  }
+
+  private RecognitionConfig.AudioEncoding getAudioEncoding() {
+    RecognitionConfig.AudioEncoding encoding = RecognitionConfig.AudioEncoding.LINEAR16;
+    if (config.encoding.equalsIgnoreCase("amr")) {
+      encoding = RecognitionConfig.AudioEncoding.AMR;
+    } else if (config.encoding.equalsIgnoreCase("amr_wb")) {
+      encoding = RecognitionConfig.AudioEncoding.AMR_WB;
+    } else if (config.encoding.equalsIgnoreCase("flac")) {
+      encoding = RecognitionConfig.AudioEncoding.FLAC;
+    } else if (config.encoding.equalsIgnoreCase("mulaw")) {
+      encoding = RecognitionConfig.AudioEncoding.MULAW;
+    } else if (config.encoding.equalsIgnoreCase("ogg_opus")) {
+      encoding = RecognitionConfig.AudioEncoding.OGG_OPUS;
+    }
+    return encoding;
+  }
+
+  private void addTranscriptWithConfidence(Schema recordSchema, List<StructuredRecord> speechArray,
```

Comment:
shouldn't need to pass in the record schema. The parts field schema should be known at initialize time. Set it and use it here